# HG changeset patch
# User Rik <rik@octave.org>
# Date 1391738709 28800
#      Thu Feb 06 18:05:09 2014 -0800
# Node ID 92ec75319e736981fa93c1800aff24c5d23b876c
# Parent  07ca8b9f8cab4ca5a29c52efaa9249613a943bd3
surfl.m: normalize using dataaspectratio, not plotboxaspectratio.

* surfl.m: Calculations of normals to surface now use dataaspectratio to
determine length, rather than plotboxaspectratio.  Modified %!demo to
showcase surfl and the differences when using the light vector and
material properties inputs to surfl.

diff --git a/scripts/plot/draw/surfl.m b/scripts/plot/draw/surfl.m
--- a/scripts/plot/draw/surfl.m
+++ b/scripts/plot/draw/surfl.m
@@ -159,17 +159,17 @@ function h = surfl (varargin)
       phi = pi / 4;  # 45 degrees
       R = [cos(phi), -sin(phi), 0;
            sin(phi),  cos(phi), 0;
            0,         0,        1];
       lv = (R * vv.').';
     endif
 
     vn = get (htmp, "vertexnormals");
-    dar = get (hax, "plotboxaspectratio");
+    dar = get (hax, "dataaspectratio");
     vn(:,:,1) *= dar(1);
     vn(:,:,2) *= dar(2);
     vn(:,:,3) *= dar(3);
 
     ## Normalize vn.
     vn ./= repmat (sqrt (sumsq (vn, 3)), [1, 1, 3]);
     [nr, nc] = size (get (htmp, "zdata"));
 
@@ -201,13 +201,12 @@ endfunction
 %! surfl (X,Y,Z);
 %! shading interp;
 %! title ('surfl() with defaults');
 
 %!demo
 %! clf;
 %! [X,Y,Z] = sombrero ();
 %! colormap (copper (64));
-%! [az, el] = view ();
-%! surfl (X,Y,Z, [az+225,el], [0.2 0.6 0.4 25]);
+%! surfl (X,Y,Z, [62.50,30], [0.2 0.6 0.4 25]);
 %! shading interp;
 %! title ('surfl() with lighting vector and material properties');
 
